phoenix-core/src/main/java/org/apache/phoenix/compile/ExpressionCompiler.java

```diff
@@ -202,6 +202,18 @@ public Expression visitLeave(ComparisonParseNode node, List<Expression> children
         ParseNode rhsNode = node.getChildren().get(1);
         Expression lhsExpr = children.get(0);
         Expression rhsExpr = children.get(1);
+
+        PDataType dataTypeOfLHSExpr = lhsExpr.getDataType();
+        if (dataTypeOfLHSExpr != null && !dataTypeOfLHSExpr.isEqualitySupported()) {
```

Comment:
It is not just equality right ? Any comparison operator is not supported for JSON data type. Right ?

Reply:
IsComparisonSuported sounds a better name in my opinion

Reply:
Makes sense. Done